src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp

```diff
@@ -99,8 +101,13 @@ class debug_configuration {
     int serialize_compile;                      // Serialize creating primitives and compiling kernels
     std::vector<std::string> forced_impl_types; // Force implementation type either ocl or onednn
     int max_kernels_per_batch;                  // Maximum number of kernels in a batch during compiling kernels
+    std::set<int64_t> dump_iteration;         // Dump n-th execution of network.
+    static std::map<uint32_t, int64_t> current_iteration;  // key: program id
```

Comment:
I think we'd better to have this current iteration in each network instead of debug config. 
Because 
1) multiple network (= stream) will have same program, but process different iteration 
2) if each network has the iteration count then we dont need this mutex 
3) Also we may need to dump iteration per network (stream), not per program.

Reply:
We need to discuss it.

Reply:
DumpIterationPerProg variable has added. Default is dump per network.

Reply:
I think PerProg dump is not necessary... While it adds complexity, I don't think it will be useful.

Reply:
Removed